### PR TITLE
Fix: Potential Data Leakage in Quantum Data Tutorial.

### LINKED ISSUE #828: Possible data leakage in quantum/docs/tutorials /quantum_data.ipynb
Hi all,

I think there is a possible [data leakage](https://en.wikipedia.org/wiki/Leakage_(machine_learning)) in [quantum data tutorial](https://github.com/tensorflow/quantum/blob/master/docs/tutorials/quantum_data.ipynb) in https://github.com/tensorflow/quantum/blob/84872800b454de6da80d6eb7faf74244a0deeb46/docs/tutorials/quantum_data.ipynb#L783-L790

when calculating https://github.com/tensorflow/quantum/blob/84872800b454de6da80d6eb7faf74244a0deeb46/docs/tutorials/quantum_data.ipynb#L731

train and test data are concatenated in these calculations. I think it is wrong, and those should be separately calculated for each.

### PR DESCRIPTION
**Pull request by OkuyanBoga**
A solution to potential data leakage in #828.

Instead of concatenating train and test sets, they should be separately dealt with when getting a stilted dataset:

In lines [L745-752](https://github.com/OkuyanBoga/quantum/blob/61da8ecf69a371999845b214f20d22b20a948a9d/docs/tutorials/quantum_data.ipynb#L868-L869):
```
y_train_new = get_stilted_dataset(S_pqk, V_pqk, S_original, V_original)
y_test_new = get_stilted_dataset(S_pqk_test, V_pqk_test, S_test_original, V_test_original)
```

where [spectrum is calculated separately for test set](https://github.com/OkuyanBoga/quantum/blob/61da8ecf69a371999845b214f20d22b20a948a9d/docs/tutorials/quantum_data.ipynb#L802-L809):
```
S_pqk_test, V_pqk_test = get_spectrum(
    tf.reshape(x_test_pqk, [-1, len(qubits) * 3]))

S_test_original, V_test_original = get_spectrum(
    tf.cast(x_test, tf.float32), gamma=0.005)

print('Eigenvectors of pqk kernel matrix for test:', V_pqk_test)
print('Eigenvectors of original kernel matrix for test:', V_test_original)
```

Closes #828.

### CODE REVIEW

**Comment by mhucka** [comment]
@OkuyanBoga Thank you for this contribution. Would you be able to resolve the (simple) conflicts that have arisen? I will then review the PR.

**Comment by OkuyanBoga** [comment]
Hi, apologies for the late response. I believe I fixed the conflicts. I hope my fix doesn't contradict the paper directly, since I cannot reproduce the results after the data leakage fix. Any opinions or ideas?

**Comment by mhucka** [comment]
Following up on my comments in the parent issue: the behavior is as intended, so this PR can be closed.
